master/internal/stream/system.go

```diff
@@ -368,31 +395,32 @@ func startup[T stream.Msg, S any](
 
 	// Scan for historical msgs matching newly-added subscriptions.
 	var newmsgs []*websocket.PreparedMessage
-	newmsgs, err = state.CollectStartupMsgs(known, *spec, ctx)
+	newmsgs, err = state.CollectStartupMsgs(ctx, known, *spec)
 	if err != nil {
 		return nil, err
 	}
 	msgs = append(msgs, newmsgs...)
 	return msgs, nil
 }
 
-func (ss *SubscriptionSet) Startup(startupMsg StartupMsg, ctx context.Context) (
+// Startup handles starting up the Subscription's in the SubscriptionSet.
```

Comment:
this is a nit, but apostrophes are for possession.

If you want to keep the word `Subscription` as an unmodified identifier, you can use the word "objects" as in "starting the Subscription objects".  But the apostrophe thing is a little confusing I think.